engine/engine_test.go

```diff
@@ -273,6 +273,115 @@ func TestEngine_Directories(t *testing.T) {
 	})
 }
 
+func TestEngine_Watch(t *testing.T) {
+	t.Run("valid single-read query", func(t *testing.T) {
+		testEnv(t, func(e Engine) {
+			query := q.KeyValue{
+				Key: q.Key{
+					Directory: q.Directory{q.String("users")},
+					Tuple:     q.Tuple{q.String("john")},
+				},
+				Value: q.Variable{q.StringType},
+			}
+
+			watch, err := e.Watch(query)
```

Comment:
To have this watch return, you will need to have a separate goroutine write to the key being watched. Change this test so you start a goroutine which writes to that key every 100ms. Make sure to synchronize the goroutine with the test and stop it after the watch returns.